pkg/sdk/client_integration_test.go

```diff
@@ -132,3 +135,72 @@ func TestClient_NewClientDriverLoggingLevel(t *testing.T) {
 		assert.Equal(t, "trace", gosnowflake.GetLogger().GetLogLevel())
 	})
 }
+
+func TestClient_AdditionalMetadata(t *testing.T) {
+	client := defaultTestClient(t)
+
+	// needed for using information_schema
+	databaseId := randomAccountObjectIdentifier()
+	require.NoError(t, client.Databases.Create(context.Background(), databaseId, &CreateDatabaseOptions{}))
+	t.Cleanup(func() {
+		require.NoError(t, client.Databases.Drop(context.Background(), databaseId, &DropDatabaseOptions{}))
+	})
+
+	metadata := map[string]string{
+		"version": "v1.0.0",
+		"method":  "create",
+	}
+
+	assertQueryMetadata := func(t *testing.T, queryId string) {
+		t.Helper()
+		result, err := client.QueryUnsafe(context.Background(), fmt.Sprintf("SELECT QUERY_ID, QUERY_TEXT FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY(RESULT_LIMIT => 2)) WHERE QUERY_ID = '%s'", queryId))
+		require.NoError(t, err)
+		require.Len(t, result, 1)
+		require.Equal(t, queryId, *result[0]["QUERY_ID"])
+		var parsedMetadata map[string]string
+		queryText := (*result[0]["QUERY_TEXT"]).(string)
+		queryMetadata := strings.Split(queryText, fmt.Sprintf("--%s", DashboardTrackingPrefix))[1]
```

Comment:
nit: both `toTrackingComment` and `fromTrackingComment` could be extracted as prod code functions (I mean, if we have prod function to create a comment we could also have the method to do it the other way round)

Reply:
moved to `tracking` package